sdk/resourcemanager/Azure.ResourceManager.Core/src/ApiVersions.cs

```diff
@@ -89,27 +89,48 @@ internal async Task<string> LoadApiVersionAsync(ProvidersOperations providers, R
                     return type.ApiVersions[0];
                 }
             }
-            return string.Empty;
+            return null;
+        }
+
+        internal string TyrGetApiVersion(ProvidersOperations providers, ResourceIdentifier resourceId, CancellationToken cancellationToken)
+        {
+            string val;
+            if (TryGetApiVersion(resourceId, out val))
+            {
+                return val;
+            }
+            return LoadApiVersion(providers, resourceId, cancellationToken);
+        }
+
+        internal async Task<string> TyrGetApiVersionAsync(ProvidersOperations providers, ResourceIdentifier resourceId, CancellationToken cancellationToken)
+        {
+            string val;
+            if (TryGetApiVersion(resourceId, out val))
+            {
+                return val;
+            }
+            return await LoadApiVersionAsync(providers, resourceId, cancellationToken).ConfigureAwait(false);
         }
 
         /// <summary>
         /// Gets the API version give a resource ID if it exist, else will return null.
         /// </summary>
         /// <returns> API version string. </returns>
-        public string TryGetApiVersion(string resourceId)
+        public bool TryGetApiVersion(string resourceId, out string val)
         {
             PropertyWrapper propertyWrapper;
             if (_loadedResourceToApiVersions.TryGetValue(resourceId, out propertyWrapper))
             {
-                return propertyWrapper.Info.GetValue(propertyWrapper.PropertyObject).ToString();
+                val = propertyWrapper.Info.GetValue(propertyWrapper.PropertyObject).ToString();
```

Comment:
I would have expected this line to look more like

val = propertyWrapper.GetValue();

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/19437#discussion_r598988753